File: Density/app/src/main/java/org/cornelldti/density/density/util/ColorBarDataSet.java
package org.cornelldti.density.density.util;

import com.github.mikephil.charting.data.BarDataSet;
import com.github.mikephil.charting.data.BarEntry;

import java.util.List;

public class ColorBarDataSet extends BarDataSet {

    public ColorBarDataSet(List<BarEntry> y_vals, String label) {
        super(y_vals, label);
    }
    @Override
    public int getColor(int index) {
        if(getEntryForIndex(index).getY() >= 0.75) {
            return getColors().get(3);
        }
        else if(getEntryForIndex(index).getY() >= 0.5 ) {
            return getColors().get(2);
        }
        else if(getEntryForIndex(index).getY() >= 0.25) {
            return getColors().get(1);
        }
        else {
            return getColors().get(0);
        }
    }

}

File: Density/app/src/main/java/org/cornelldti/density/flux/colorbarutil/ColorBarDataSet.kt
package org.cornelldti.density.flux.colorbarutil

import com.github.mikephil.charting.data.BarDataSet
import com.github.mikephil.charting.data.BarEntry

class ColorBarDataSet(y_vals: List<BarEntry>, label: String) : BarDataSet(y_vals, label) {

    // We need this dummy override to make Kotlin compiler happy. Otherwise there will be a
    // collision in this method in the Java code that we can't control.
    @Suppress("RedundantOverride")
    override fun getEntryIndex(e: BarEntry): Int = super.getEntryIndex(e)

    override fun getColor(index: Int): Int {
        return if (getEntryForIndex(index).y >= 0.75) {
            colors[3]
        } else if (getEntryForIndex(index).y >= 0.5) {
            colors[2]
        } else if (getEntryForIndex(index).y >= 0.25) {
            colors[1]
        } else if (getEntryForIndex(index).y > 0) {
            colors[0]
        } else {
            colors[4]
        }
    }

}
